### PR TITLE
Feature/some code cleanup for VM functions

### PR DESCRIPTION
**Pull request by JaXt0r**
Basically a code cleanup feature. I simplified the VMExternal calls and added NPC inventory. But not visible right now, just the plain functions.

There's no "External is missing" warning any longer when loading NPCs from OC.

@JucanAndreiDaniel Please do a code review mostly. Sorry for the amount of changes. This is also the reason to have it in main soon before it blows up even further.

# To test

## What changed code wise:
* Perception enums added
* NPC inventory filled (code wise)
* External functions have no "UnityEvent" any longer. This eases the implementation and removes complexity (as we never had more than 1 subscriber)
* SpawnPoints are now checked from WayPoints and FreePoints (WayNetManager.cs; it means we finally can spawn Meatbugs inside OC - please check)
* Added all the Thorus daedalus external functions which are used for Thorus ZM_Boss() routine. Mostly mocks, but no errors warnings are thrown any longer

## What to test:
* Do the NPCs dance? (Except Thorus)
* Do one Meatbug inside OC has a walking animation loop?
* Any new Exceptions? - If not, we're good to merge.

# Checklist
(Please judge for yourself which entry is valuable for your PR)

## Scene
* [x] FeatureFlags reverted to right value

# Testing
* [x] Merged _main_ into this branch and tested with the latest features
* [ ] Tested with PCVR
* [ ] Tested with Pico / Quest

### CODE REVIEW

File: Assets/GothicVR/Scripts/Phoenix/Interface/Vm/VmGothicExternals.cs [hunk 1]
```diff
@@ -0,0 +1,456 @@
...
+        [MonoPInvokeCallback(typeof(PxVm.PxVmExternalCallback))]
+        public static void Wld_InsertNpc(IntPtr vmPtr)
+        {
+            var spawnpoint = PxVm.VmStackPopString(vmPtr);
+            var npcInstance = PxVm.pxVmStackPopInt(vmPtr);
+            
+            NpcCreator.I.ExtWldInsertNpc(npcInstance, spawnpoint);
+        }
+
+        public static void Wld_IsFPAvailable(IntPtr vmPtr)
```
Comment (JucanAndreiDaniel) [inline Assets/GothicVR/Scripts/Phoenix/Interface/Vm/VmGothicExternals.cs:189]: This function is missing the callback

```suggestion
        [MonoPInvokeCallback(typeof(PxVm.PxVmExternalCallback))]
        public static void Wld_IsFPAvailable(IntPtr vmPtr)
```

File: Assets/GothicVR/Scripts/Manager/GvrSceneManager.cs [hunk 2]
```diff
@@ -53,18 +54,30 @@ public async Task LoadStartupScenes()
...
+            else
+                debugFreshlyDoneLoading = false;
+            
+            if (FeatureFlags.I.CreateOcNpcs)
+                PxVm.CallFunction(GameData.I.VmGothicPtr, "STARTUP_SUB_OLDCAMP");
+
+            if (FeatureFlags.I.CreateDebugIdleAnimations)
+                NpcCreator.I.DebugAddIdleAnimationToAllNpc();
+        }
+
```
Comment (JucanAndreiDaniel) [inline Assets/GothicVR/Scripts/Manager/GvrSceneManager.cs:80]: NPCs are loaded only if we directly load into the world, but not if we go through the main menu, is this intended?

**Review by JucanAndreiDaniel — CHANGES_REQUESTED** [review]
Small and easy to fix issues.
Please also remove the duplicate teleport menu from the General scene in the vr player UI

**Comment by JaXt0r** [comment]
@JucanAndreiDaniel - I fixed these things. Especially the missing MonoPInvokeCallback would've been annoying. Thanks for looking closely!

**Review by JucanAndreiDaniel — APPROVED** [review]
LGTM now

**Pull request merged**